### DISCUSSION TITLE
Are there wrappers for tree cursor APIs?

### DISCUSSION THREAD

**Discussion by champignoom** [nvim-treesitter/nvim-treesitter#4817] [Q&A]
Is there (going to be) lua wrappers for [tree cursor APIs](https://tree-sitter.github.io/tree-sitter/using-parsers#walking-trees-with-tree-cursors)?

```c
TSTreeCursor ts_tree_cursor_new(TSNode);

bool ts_tree_cursor_goto_first_child(TSTreeCursor *);
bool ts_tree_cursor_goto_next_sibling(TSTreeCursor *);
bool ts_tree_cursor_goto_parent(TSTreeCursor *);

TSNode ts_tree_cursor_current_node(const TSTreeCursor *);
const char *ts_tree_cursor_current_field_name(const TSTreeCursor *);
TSFieldId ts_tree_cursor_current_field_id(const TSTreeCursor *);
```

**Comment by clason** [comment]
Wrong repo. Treesitter API is handled by Neovim.

(And yes, if someone writes them. You could start by looking at https://github.com/euclidianAce/ltreesitter.)

**Comment by lewis6991** [comment]
We already have `node:iter_children()`, `node:next_child()` and `node:parent()`. What would `TSTreeCursor` API's allow that cannot be done with these?
